modules/core/src/v2/coins/utxo/recovery/crossChainRecovery.ts

```diff
@@ -41,607 +52,354 @@ class BitgoPublicApi {
+  unspents.forEach((unspent) => {
+    utxolib.bitgo.addToTransactionBuilder(transactionBuilder, unspent);
+  });
+  let transaction = transactionBuilder.buildIncomplete();
+  if (signer) {
+    transaction = signAndVerifyWalletTransaction(transactionBuilder, unspents, signer, { isLastSignature: false });
   }
+  return transaction;
+}
 
-  async buildUnsigned(): Promise<CrossChainRecoveryUnsigned> {
-    if (_.isUndefined(this.txInfo)) {
-      throw new Error('Could not find txInfo. Please build a transaction');
-    }
-    const incomplete = this.recoveryTx.buildIncomplete();
-
-    const txInfo: any = {
-      nP2SHInputs: 0,
-      nSegwitInputs: 0,
-    };
-
-    for (const input of this.txInfo.inputs) {
-      if (input.chain === 10 || input.chain === 11) {
-        txInfo.nSegwitInputs++;
-      } else {
-        txInfo.nP2SHInputs++;
-      }
-    }
-
-    txInfo.nOutputs = 1;
-    txInfo.unspents = _.map(
-      this.txInfo.inputs,
-      _.partialRight(_.pick, ['chain', 'index', 'redeemScript', 'id', 'address', 'value'])
-    );
-    txInfo.changeAddresses = [];
-    txInfo.walletAddressDetails = {};
-
-    const feeInfo: any = {};
+function getTxInfo(
+  transaction: utxolib.bitgo.UtxoTransaction,
+  unspents: WalletUnspent[],
+  walletId: string,
+  walletKeys: RootWalletKeys
+): TransactionInfo {
+  const inputAmount = utxolib.bitgo.unspentSum(unspents);
+  const outputAmount = transaction.outs.reduce((sum, o) => sum + o.value, 0);
+  const outputs = transaction.outs.map((o) => ({
+    address: utxolib.address.fromOutputScript(o.script, transaction.network),
+    valueString: o.value.toString(),
+    change: false,
+  }));
+  const inputs = unspents.map((u) => {
+    // NOTE:
+    // The `redeemScript` and `walletScript` properties are required for legacy versions of BitGoJS
+    // which might require these scripts for signing. The Wallet Recovery Wizard (WRW) can create
+    // unsigned prebuilds that are submitted to BitGoJS instances which are not necessarily the same
+    // version.
+    const addressKeys = walletKeys.deriveForChainAndIndex(u.chain, u.index);
+    const scriptType = scriptTypeForChain(u.chain);
+    const { redeemScript, witnessScript } = outputScripts.createOutputScript2of3(addressKeys.publicKeys, scriptType);
 
-    feeInfo.size =
-      VirtualSizes.txOverheadSize +
-      VirtualSizes.txP2shInputSize * this.txInfo.inputs.length +
-      VirtualSizes.txP2pkhOutputSize;
+    return {
+      ...u,
+      wallet: walletId,
+      fromWallet: walletId,
+      redeemScript: redeemScript?.toString('hex'),
+      witnessScript: witnessScript?.toString('hex'),
+    } as WalletUnspentLegacy;
+  });
+  return {
+    inputAmount,
+    outputAmount,
+    minerFee: inputAmount - outputAmount,
+    spendAmount: outputAmount,
+    inputs,
+    unspents: inputs,
+    outputs,
+    externalOutputs: outputs,
+    changeOutputs: [],
+    payGoFee: 0,
+  } as TransactionInfo;
```

Comment:
Why is this cast necessary? I'm also a bit confused as to why there are so many of these fields aren't defined within the `TransactionInfo` interface.

Reply:
I kept these fields to stay backwards compatible. I don't think they are required by any methods these days but they may be needed by legacy consumers of this. Will add a comment.